tests/unit/igraph_widest_paths.c

```diff
@@ -0,0 +1,587 @@
+        free(VECTOR(*vertices)[i]);
+        igraph_vector_int_destroy(VECTOR(*edges)[i]);
+        free(VECTOR(*edges)[i]);
+    }
+    igraph_vector_ptr_destroy(edges);
+    igraph_vector_ptr_destroy(vertices);
+}
+
+/* destroy all structures */
+void destroy_all(igraph_t *g, igraph_vector_t *w, igraph_matrix_t *res1, igraph_matrix_t *res2,
+                    igraph_vs_t *from, igraph_vs_t *to,
+                    igraph_vector_ptr_t *vertices, igraph_vector_ptr_t *edges,
+                    igraph_vector_int_t *predecessors, igraph_vector_int_t *inbound_edges,
+                    igraph_vector_int_t *vertices2, igraph_vector_int_t *edges2) {
+
+    destroy_vertices_and_edges(vertices, edges, predecessors, inbound_edges, vertices2, edges2);
+
+    igraph_vs_destroy(to);
+    igraph_vs_destroy(from);
+    igraph_matrix_destroy(res2);
+    igraph_matrix_destroy(res1);
+
+    igraph_vector_destroy(w);
+    igraph_destroy(g);
+}
+
+/* print results of just the matrices */
+void check_and_print_matrices(igraph_matrix_t *res1, igraph_matrix_t *res2) {
+    IGRAPH_ASSERT(igraph_matrix_all_e(res1, res2));
+    print_matrix_format(res1, stdout, "%f");
+}
+
+/* prints results of all widest path algorithms */
+void print_results(igraph_integer_t n, igraph_matrix_t *res1, igraph_matrix_t *res2,
+                    igraph_vector_ptr_t *vertices, igraph_vector_ptr_t *edges,
+                    igraph_vector_int_t *predecessors, igraph_vector_int_t *inbound_edges,
+                    igraph_vector_int_t *vertices2, igraph_vector_int_t *edges2) {
+    igraph_integer_t i;
+
+    check_and_print_matrices(res1, res2);
+    printf("\n");
+
+    for (i = 0; i < n; i++) {
+        printf("path to node %" IGRAPH_PRId ":\n", i);
+        igraph_vector_int_t *vertex_path = VECTOR(*vertices)[i];
+        igraph_vector_int_t *edge_path = VECTOR(*edges)[i];
+        printf("  vertices: ");
+        print_vector_int(vertex_path);
+        printf("  edges:    ");
+        print_vector_int(edge_path);
+    }
+    printf("\n");
+
+    printf("predecessors:  ");
+    print_vector_int(predecessors);
+    printf("inbound_edges: ");
+    print_vector_int(inbound_edges);
+    printf("\n");
+
+    printf("vertex path:   ");
+    print_vector_int(vertices2);
+    printf("edge path:     ");
+    print_vector_int(edges2);
+    printf("\n");
+}
+
+/* confirm that all widest paths algorithms fail */
+void check_invalid_input(igraph_t *g, igraph_vector_t *w, igraph_matrix_t *res1,
+                        igraph_vs_t *from, igraph_vs_t *to,
+                        igraph_vector_ptr_t *vertices, igraph_vector_ptr_t *edges,
+                        igraph_vector_int_t *predecessors, igraph_vector_int_t *inbound_edges,
+                        igraph_vector_int_t *vertices2, igraph_vector_int_t *edges2) {
+    CHECK_ERROR(igraph_widest_paths_dijkstra(/* graph */ g, res1,
+                            *from, *to, /* weights */ NULL,
```

Comment:
Yep you're 100% right. Looking again, I found that, as well as even more silly mistakes that were left in the code, probably because I rushed the last couple commits (my fault, sorry!). It'll be a couple days before I can really go over the test outputs properly and make sure I didn't do anything else silly.